lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2582,20 +2604,25 @@ bool AvailabilityWalker::diagAvailability(const ValueDecl *D, SourceRange R,
   if (!isAccessorWithDeprecatedStorage)
     TC.diagnoseIfDeprecated(R, DC, D, call);
 
-  if (AllowPotentiallyUnavailableProtocol && isa<ProtocolDecl>(D))
+  if (Flags.contains(DeclAvailabilityFlag::AllowPotentiallyUnavailableProtocol)
+        && isa<ProtocolDecl>(D))
+    return false;
+
+  if (Flags.contains(DeclAvailabilityFlag::AllowPotentiallyUnavailable))
```

Comment:
Nitpick: I think it makes sense to check these in the opposite order, to avoid doing work.